Sources/Logging/Logging.swift

```diff
@@ -251,34 +251,53 @@ extension Logger {
     }
 }
 
+/// The `LogHandlerFactory` is the bridge between the `LoggingSystem` and the metrics backend implementation.
+/// `LogHandlerFactory` role is to initialize concrete `LogHandler` instances,
+/// it may also contain any additional state it wishes to keep for operating those handlers if it needs to.
+///
+/// This type is an implementation detail and should not be used directly, unless implementing your own metrics backend.
+/// To use the Logging API, please refer to the documentation of `LoggingSystem`.
+public protocol LogHandlerFactory {
```

Comment:
why do we need a protocol if it's just one func?

Reply:
Reasons are a bit other than "we really need one to (store / do) X" I guess

- as PR title says, init style alignment between the metrics and logging APIs -- was the main motivation
- code structure "hint" to people implementing logging backends (that's a weaker argument I guess, though hopefully would result in uniform looking "this is how you init library X and this is how library Y")

Reply:
@ktoso couldn't we change metrics instead? This would break major version here

Reply:
also unconvinced a protocol is the right choice here

Reply:
> @ktoso couldn't we change metrics instead? This would break major version here

Metrics definitely need a "factory" type though, what's your idea?

> also unconvinced a protocol is the right choice here

Open to ideas :)